parsec/core/fs/workspacefs/workspacefs.py

```diff
@@ -186,12 +186,55 @@ def workspace_name(self) -> str:
     async def move(self, source: AnyPath, destination: AnyPath):
         source = FsPath(source)
         destination = FsPath(destination)
+        real_destination = destination
         if source.parent == destination.parent:
             return await self.rename(source, destination)
-        # TODO - reference implementation:
-        # https://github.com/python/cpython/blob/3.7/Lib/shutil.py#L525
+        try:
+            if await self.is_dir(destination):
+                real_destination = destination.joinpath(source.name)
+                if await self.path_info(real_destination):
+                    raise FileExistsError
+        # If real_destination is not found, we can continue
+        except FileNotFoundError as e:
+            pass
+        if await self.is_dir(source):
+            await self.copytree(source, real_destination)
+            await self.rmtree(source)
+            return
+        elif await self.is_file(source):
+            await self.copyfile(source, real_destination)
+            await self.entry_transactions.file_delete(source)
+            return
         raise NotImplementedError
 
+    def _destinsrc(src: AnyPath, dst: AnyPath):
+        try:
+            dst.relative_to(src)
+            return True
+        except ValueError:
+            return False
+
+    async def copytree(self, source_path: AnyPath, target_path: AnyPath):
+        source_files = await self.listdir(source_path)
+        await self.mkdir(target_path)
+        errors = []
```

Comment:
This list seems to be unused

Reply:
Yes, I forgot it, it was part of my first implementation where I kept a list of failed copies and raised an error only at the end, mimicking shutil behaviour. Do you want me to re-implement it that way?